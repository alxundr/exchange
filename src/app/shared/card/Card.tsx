import React from 'react';
import './Card.scss';

export enum CardColor {
  Default = 'white',
  Dark = '#d6d6d6'
}

export interface CardProps {
  color?: CardColor;
  style?: React.CSSProperties,
}

const Card: React.FC<CardProps> = ({ children, style, color = CardColor.Default }) => {
  return (
    <div className="card" style={{...style, backgroundColor: color}}>
      {children}
    </div>
  )
}

export default Card;
