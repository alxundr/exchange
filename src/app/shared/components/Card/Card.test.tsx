import React from "react";
import Card from "./Card";
import { render } from "@testing-library/react";

describe("Card", () => {
  test("matches snapshot", () => {
    const card = render(<Card style={{ color: "white", backgroundColor: "black" }}>Luke, I am your father!</Card>);
    expect(card.container).toMatchSnapshot();
  });
});
