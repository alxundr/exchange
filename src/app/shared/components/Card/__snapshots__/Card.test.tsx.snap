// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<div>
  <div
    class="card"
    style="color: white; background-color: white;"
  >
    Luke, I am your father!
  </div>
</div>
`;
